src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -10,6 +10,7 @@ namespace Microsoft.AspNet.Server.Kestrel.Infrastructure
 {
     public struct MemoryPoolIterator2
     {
+        private const int _maxHeaderStackLength = 16384;
```

Comment:
This value is in the wrong layer MemoryPoolIterator2 doesn't understand what it's looking at. That's the caller's job.